sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DefaultSynchronousMetricStorage.java

```diff
@@ -217,6 +237,6 @@ public MetricDescriptor getMetricDescriptor() {
   private static class AggregatorHolder<T extends PointData, U extends ExemplarData> {
     private final ConcurrentHashMap<Attributes, AggregatorHandle<T, U>> aggregatorHandles =
         new ConcurrentHashMap<>();
-    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final AtomicInteger recordsInProgress = new AtomicInteger(0);
```

Comment:
(maybe a future rename `AggregatorHolder` to something with `..Interval..` could help with visualizing this better?)
```suggestion
    // recording threads grab the current interval (AggregatorHolder) and atomically increment
    // this by 2 before recording against it (and then decrement by two when done)
    //
    // the collection thread grabs the current interval (AggregatorHolder) and atomically
    // increments this by 1 to "lock" this interval (and then waits for any active recording
    // threads to complete before collecting it)
    //
    // recording threads check the return value of their atomic increment, and if it's odd
    // that means the collector thread has "locked" this interval for collection
    //
    // but before the collector "locks" the interval it sets up a new current interval
    // (AggregatorHolder), and so if a recording thread enounters an odd value,
    // all it needs to do is release the "read lock" it just obtained (decrementing by 2),
    // and then grab and record against the new current interval (AggregatorHolder)
    private final AtomicInteger activeRecordingThreads = new AtomicInteger(0);
```